### PR TITLE
Integrate LoanRecords with Person class and storage mechanism

### LINKED ISSUE #37: Integrate LoanRecords class with Person class and storage mechanism

### PR DESCRIPTION
**Pull request by Joseph31416**
fixes #37 

DateUtil class is added to handle String to Date and vice versa formatting operations.

Do take note of the date format. This is the expected format for the date string to be parsed into a Date object.
`private static final String DATE_FORMAT = "yyyy-MM-dd";`

Each Person object has a distinct LoanRecords object associated with them.

The id field in LoanRecords has been removed as it serves no practical purpose. Instead, a new nextLoanId field is introduced in LoanRecords to keep track of the nextLoanId to issue when another loan is added to the person. 

Take note of the overloaded `addLoan` method:
`public void addLoan(float amount, String startDate, String endDate)`
`public void addLoan(float amount, Date startDate, Date endDate)`

The first one should be more convenient to be used. Just ensure that the startDate complies with the DATE_FORMAT as given in the DateUtil class then things will work. Validation checks have been put in place so an IllegalValueException will be thrown if the format is violated. 
A future TODO: check that startDate comes before endDate

The Loan and LoanRecords have been made to be serialisable. Details are in the `JsonAdaptedLoan.java` and `JsonAdaptedLoanRecords.java` file. The key point is that the nextLoanId is stored in each serialised LoanRecords object.

### CODE REVIEW

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -4,39 +4,54 @@
...
         return new Person[] {
             new Person(new Name("Alex Yeoh"), new Phone("87438807"), new Email("alexyeoh@example.com"),
                 new Address("Blk 30 Geylang Street 29, #06-40"),
-                getTagSet("friends")),
+                getTagSet("friends"),
+                tempLoanRecords),
             new Person(new Name("Bernice Yu"), new Phone("99272758"), new Email("berniceyu@example.com"),
                 new Address("Blk 30 Lorong 3 Serangoon Gardens, #07-18"),
-                getTagSet("colleagues", "friends")),
+                getTagSet("colleagues", "friends"),
```
Comment (xiaorui-ui) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:37]: Are tags complusory?

**Review by xiaorui-ui — COMMENTED** [review]
Lgtm! Waiting for Jun Wei's response

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 1 above)
Comment (Joseph31416) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:37]: This is from the default sample given by them, so I don't think there's a need to change it. Only relevant change to this is the addition of LoanRecords entry for each sample person.

File: src/main/java/seedu/address/model/person/LoanRecords.java [hunk 2]
```diff
@@ -39,27 +55,33 @@ public void addLoan(Loan loan) {
      * @param endDate A valid end date.
      */
     public void addLoan(float amount, Date startDate, Date endDate) {
-        Loan loan = new Loan(amount, startDate, endDate);
+        Loan loan = new Loan(nextLoanId, amount, startDate, endDate);
```
Comment (kjw142857) [inline src/main/java/seedu/address/model/person/LoanRecords.java:58]: As LoanRecords are created by person, is it possible for two loans to two different people to have the same id?

**Review by kjw142857 — COMMENTED** [review]
Thanks for implementing the LoanRecords for future usage! Just have one small thing to clarify with regards to the id (you can see my comment below). Other than that, LGTM!

**Review by kjw142857 — APPROVED** [review]

**Pull request merged**